django/applications/catmaid/control/skeletonexport.py

```diff
@@ -1164,19 +1164,18 @@ def __init__(self, parent_id, x, y, z):
     class Skeleton():
         def __init__(self):
             self.nodes = {}  # type: Dict[Any, Node]
-            self.raw_cable = 0
-            self.smooth_cable = 0
-            self.principal_branch_cable = 0
+            self.raw_cable = 0.0
+            self.smooth_cable = 0.0
+            self.principal_branch_cable = 0.0
             self.n_ends = 0
             self.n_branch = 0
             self.n_pre = 0
             self.n_post = 0
 
-    skeletons = defaultdict(dict)  # type: DefaultDict[Any, Dict]
-                                   # skeleton ID vs (node ID vs Node)
+    skeletons = {} # type: Dict[Any, Skeleton]
```

Comment:
Worth looking carefully at this - I think this makes the logic simpler. I hope it's correct.

Reply:
This change from `defaultdict(dict)` to `dict` seems correct to me. The `defaultdict(dict)` wasn't really used as such before, so it's good the use fits the definition now.

I might be wrong, but I don' think the change in lines 1178 and 1179 work (see below).